apps/lionsbrgstatus/lions_gate_status.star

```diff
@@ -0,0 +1,86 @@
+ARROWS_API_URL = "https://b60n09kp22.execute-api.us-west-2.amazonaws.com/prod/getarrows"
+ARROWS_API_KEY = secret.decrypt("AV6+xWcEjAB6ZacmRADoIGkvOwLus8+PfPTFHmkQRkwAa581Du8eIsNYWUwMkuK9epGXh9wmfhfoBKGsY/pN2PCtRVHgeZawcXk5+UOG57CjyFm1jftqzJPFIhtebnRYBmb5wRgwXnGgS8e79Mg9DWs6fzRtaLQb1f8Fz9PZiPtYCvLDd9Srfa8EUqeRHQ==")
+
+def fetch_arrow_directions():
+    headers = {"x-api-key": ARROWS_API_KEY}
+    r = http.get(ARROWS_API_URL, headers = headers, ttl_seconds = 30)
+    if r.status_code == 200:
+        return r.json().get("arrow_directions", [])
+    else:
+        return []
+
+def fetch_dms_data():
+    r = http.get(API_URL)
+    if r.status_code == 200:
+        return r.json()
+    else:
+        return None
+
+def get_sign_by_signNo(dms_data, sign_number):
+    for sign in dms_data:
+        if sign["location"]["signNo"] == sign_number:
+            return sign
+    return None
+
+def get_image_src_for_direction(direction):
+    """Return the appropriate image src based on arrow direction."""
+    if direction == "back":
+        return lane_back
+    elif direction == "forward":
+        return lane_forward
+    elif direction == "closed":
+        return lane_closed
+    else:
+        return lane_closed  # default, can be changed as per requirement
+
+def render_dms():
+    dms_data = fetch_dms_data()
+    arrow_directions = fetch_arrow_directions()
+    dms_sign = get_sign_by_signNo(dms_data, sign_number)
+    decoded_text = base64.decode(dms_sign["location"]["content"]["pages"][0]["lines"][0]["text"])
+    decoded_text = decoded_text.replace("[nl]  ", " - ")
+    sign_text = decoded_text.replace("[pt30o2]  ", "")
+
+    # Build lane icons list iteratively
+    lane_icons = []
+    for direction in arrow_directions:
+        lane_icons.append(render.Image(src = get_image_src_for_direction(direction), width = 10))
+        lane_icons.append(render.Box(width = 11))
+
+    # Remove the last gap after the last icon
+    lane_icons.pop()
```

Comment:
If `arrow_directions` array is empty (which it can be if you get anything other than 200 response code from the aws API call), then lane_icons will be empty and `.pop` will throw `index -1 out of range: empty list` exception. Might be worth making this more robust or failing in a more expected way if the API call to get the lane directions fails.